src/seb/cli/table.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional
+
+import pandas as pd
+from rich.console import Console
+from rich.table import Table
+
+import seb
+from seb.registries import get_all_models
+
+
+def get_main_score(task: seb.TaskResult, langs: Optional[list[str]]) -> float:
+    if langs is None:
+        _langs = task.languages
+    else:
+        _langs = set(langs) & set(task.languages)
+    return task.get_main_score(_langs) * 100
+
+
+def open_source_to_string(open_source: bool) -> str:
+    return "✓" if open_source else "✗"
+
+
+def benchmark_result_to_row(
+    result: seb.BenchmarkResults,
+    langs: Optional[list[str]],
+) -> pd.DataFrame:
+    mdl_name = result.meta.name
+    # sort by task name
+    task_results = result.task_results
+    sorted_tasks = sorted(task_results, key=lambda t: t.task_name)
+    task_names = [t.task_name for t in sorted_tasks]
+    scores = [get_main_score(t, langs) for t in sorted_tasks]  # type: ignore
+
+    df = pd.DataFrame([scores], columns=task_names, index=[mdl_name])
+    df["Average Score"] = result.get_main_score()  # type: ignore
+    df["Open Source"] = open_source_to_string(result.meta.open_source)
+    df["Embedding Size"] = result.meta.embedding_size
+    return df
+
+
+def convert_to_table(
+    results: list[seb.BenchmarkResults],
+    langs: Optional[list[str]],
+) -> pd.DataFrame:
+    rows = [benchmark_result_to_row(result, langs) for result in results]
+    df = pd.concat(rows)
+    df = df.sort_values(by="Average Score", ascending=False)
+    df["Average Rank"] = compute_avg_rank(df)
+
+    # ensure that the average and open source are the first column
+    cols = df.columns.tolist()
+    first_columns = [
+        "Average Score",
+        "Average Rank",
+    ]
+    other_cols = sorted(c for c in cols if c not in first_columns)
+    df = df[first_columns + other_cols]
+    df = df.drop(columns=["Open Source", "Embedding Size"])
+
+    # convert name to column
+    df = df.reset_index()
+    df = df.rename(columns={"index": "Model"})
+    ranks = [i + 1 for i in range(len(df.index))]
+    df.insert(0, "Rank", ranks)) + 1)
+    return df
+
+
+def compute_avg_rank(df: pd.DataFrame) -> pd.Series:
+    """
+    For each model in the dataset, for each task, compute the rank of the model
+    and then compute the average rank.
+    """
+    df = df.drop(columns=["Average Score", "Open Source", "Embedding Size"])
+
+    ranks = df.rank(axis=0, ascending=False)
+    avg_ranks = ranks.mean(axis=1)
+    return avg_ranks
+
+
+def display_model_table(result: seb.BenchmarkResults, langs: Optional[list[str]]):
```

Comment:
I would suggest that the display function only takes in the dataframe

Reply:
Okay I can change that.